pkg/controller/kogitoinfra/kogitoinfra_controller.go

```diff
@@ -116,121 +92,28 @@ type ReconcileKogitoInfra struct {
 func (r *ReconcileKogitoInfra) Reconcile(request reconcile.Request) (result reconcile.Result, resultErr error) {
 	log.Infof("Reconciling KogitoInfra for %s in %s", request.Name, request.Namespace)
 
-	// Requires only one KogitoInfra instance in this namespace
-	instances := &appv1alpha1.KogitoInfraList{}
-	if err := kubernetes.ResourceC(r.client).ListWithNamespace(request.Namespace, instances); err != nil {
-		return reconcile.Result{}, err
-	}
-	if len(instances.Items) > 1 {
-		return reconcile.Result{RequeueAfter: time.Duration(5) * time.Minute},
-			fmt.Errorf("There's more than one KogitoInfra resource in this namespace, please delete one of them ")
-	}
-
 	// Fetch the KogitoInfra instance
-	instance := &appv1alpha1.KogitoInfra{}
-	if exists, err := kubernetes.ResourceC(r.client).FetchWithKey(request.NamespacedName, instance); err != nil {
-		return reconcile.Result{}, err
-	} else if !exists {
-		return reconcile.Result{}, nil
-	}
-
-	// watcher will be nil on test env
-	if controllerWatcher != nil && !controllerWatcher.AreAllObjectsWatched() {
-		if (instance.Spec.InstallKafka && !controllerWatcher.IsGroupWatched(kafkav1beta1.SchemeGroupVersion.Group)) ||
-			(instance.Spec.InstallInfinispan && !controllerWatcher.IsGroupWatched(infinispanv1.SchemeGroupVersion.Group)) ||
-			(instance.Spec.InstallKeycloak && !controllerWatcher.IsGroupWatched(keycloakv1alpha1.SchemeGroupVersion.Group)) {
-			// try to add them
-			if err := controllerWatcher.Watch(watchedObjects...); err != nil {
-				return reconcile.Result{}, err
-			}
-			if !controllerWatcher.AreAllObjectsWatched() {
-				log.Warn("Dependencies not found for KogitoInfra, please install them.")
-				return reconcile.Result{RequeueAfter: time.Minute * 5}, nil
```

Comment:
This check can be added in your reconcilers. :)

Reply:
already did